### DISCUSSION TITLE
srsRAN Project - RIC support

### DISCUSSION THREAD

**Discussion by bondadakumarsai** [srsran/srsRAN_Project#79] [General Help] [answered]
Hello all, I see that srsRAN Project supports the E2 interface. Has anyone tried connecting the Near Real-time RIC to srsRAN gNB with the e2 interface? Or if I want to connect RIC to srsRAN, does srsRAN supports it?

**Comment by pgawlowicz** [comment, marked as answer]
Hi,

Currently, only the srsRAN_4G project supports E2 interface. The current release supports only a limited version of E2SM_KPM, i.e., only Service Report Style 1 and exposes only a few test metrics. E2SM_KPM (with a lot of metrics) will be fully supported in the next release of srsRAN_4G.
Support of E2 interface in srsRAN_Project will be added in future. 

Best Regards,
Piotr

**Reply by Ali78k** [reply]
Hi @pgawlowicz ,

Which version of srsRAN_4G project does support E2 interface? Do you mean OAIC project?

Thanks,
Ali.

**Reply by pgawlowicz** [reply]
Hi,

E2 was added in our last release of srsRAN_4G project or simply master branch here: https://github.com/srsran/srsRAN_4G/tree/master

E2 agent can be enabled in config here:
https://github.com/srsran/srsRAN_4G/blob/master/srsenb/enb.conf.example#L373

But as I said the support in the current release is very limited and we are working on the extensions.

Best Regards,
Piotr

**Reply by Ali78k** [reply]
Sounds good.
Thank you.

**Reply by bondadakumarsai** [reply]
Hi, Is it possible to connect the E2 agent to Near RT RIC with the current codebase?

Best regards
Kumar

**Reply by vamaglog** [reply]
Hi, 
When you say that E2 will be added in the future, what is the roadmap for supporting of E2 interface in srsRAN_Project? 
Thank you in advance.

**Reply by pgawlowicz** [reply]
Please check our tutorial on connecting srsGNB to the FlexRIC over E2 interface - https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html
